### ISSUE TITLE
instanceof buffer in types.js returning inconsitent results in Angular with type Uint8Array when using Ng serve and build

### ISSUE THREAD

**Issue by NickDavid-code** [amqp/rhea#308]
I am running an Angular project that is using Azure Event Hubs to send and receive events with the following version:
```
Angular CLI: 8.3.27
Node: 12.13.1
OS: win32 x64
Angular: 8.2.14
```
We attempt to add eventdata to our batch using `batch.tryAdd({ body: eventData });` Based on the following https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/typescript/src/sendEvents.ts. batch.TryAdd which [here](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/src/eventDataBatch.ts#L308) calls [message.encode = function(msg) ](https://github.com/amqp/rhea/blob/1473a6567f6448768009e1220da89d617820d31c/lib/message.js#L205). Following the code execution we get here: https://github.com/amqp/rhea/blob/master/lib/types.js#L455 where we Uint8Array object as the input. When running our code with ng serve    ` } else if (o instanceof Buffer) {` is TRUE, but when building the code `    } else if (o instanceof Buffer) {` is false for the same object `Uint8Array(443)`. Is there any reason behind this inconsistent result?

**Comment by grs** [comment]
I don't know. What is different between running with ng serve and 'building the code'? Is there some extra step in the latter or a different handling of browserify or something?

Possibly the code in rhea there should test for instanceof Buffer || instance of Uint8Array.

**Comment by faraazhabeeb123** [comment]
> I don't know. What is different between running with ng serve and 'building the code'? Is there some extra step in the latter or a different handling of browserify or something?
> 
> Possibly the code in rhea there should test for instanceof Buffer || instance of Uint8Array.

What we meant here is ngserve vs ngbuild.

**Comment by NickDavid-code** [comment]
> > I don't know. What is different between running with ng serve and 'building the code'? Is there some extra step in the latter or a different handling of browserify or something?
> > Possibly the code in rhea there should test for instanceof Buffer || instance of Uint8Array.
> 
> What we meant here is ngserve vs ngbuild.

Yes using both `ng build` and `ng build --prod` we see this issue while we do not on `ng serve`. I made the changes in node_modules of my machine and ran ng build and the code is now working. Is it possible you could bump up the version number of this?

**Comment by grs** [comment]
1.0.23 now released with that fix in it
